src/main/java/io/trino/tpch/SharingTextPoolSupplier.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tpch;
+
+import javax.annotation.concurrent.GuardedBy;
+
+import java.lang.ref.SoftReference;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Supplier;
+import java.util.logging.Logger;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+final class SharingTextPoolSupplier
+        implements Supplier<TextPool>
+{
+    private static final Logger log = Logger.getLogger(SharingTextPoolSupplier.class.getName());
+
+    private final ScheduledExecutorService executor;
+    private final Supplier<TextPool> loader;
+
+    @GuardedBy("this")
+    private SoftReference<TextPool> shared = new SoftReference<>(null);
+
+    public SharingTextPoolSupplier(ScheduledExecutorService executor, Supplier<TextPool> loader)
```

Comment:
not sure it is worth to pass external executor. Not a big deal though.

Reply:
if it gets inited in ctor, who's going to close it?

(the SharingTextPoolSupplier class in itself doesn't assume there is ever single instance created)

Reply:
It could be static, daemon. But yeah - this is cleaner.

Reply:
It is static and daemon. 
Not sure why i put the field it in the other class.  Maybe i wanted the new class to be nicer than the old one?